pkgs/development/mobile/androidenv/ndk-bundle/default.nix

```
@@ -6,6 +6,7 @@ let
   runtime_paths = lib.makeBinPath (with pkgsHostHost; [
     coreutils file findutils gawk gnugrep gnused jdk python3 which
   ]) + ":${platform-tools}/platform-tools";
+  lib_paths = lib.makeLibraryPath (with pkgs; [ libcxx ]);
```

I needed to add

```suggestion
  lib_paths = lib.makeLibraryPath (with pkgs; [ libcxx libxml2 ]);
```

to get my app working.

I'm not sure if there's more than this issue, if a different bundled lib is needed.

Maybe checking `/nix/store/<androidsdk-path>/libexec/android-sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/lib/` is worth it? (I have no clue what I'm talking about, but I've found libxml2 there at least)

---

Thank you for your suggestion, but this PR has a scope defined by its issue. Please create your issue separately to clarify this PR is suitable to solve that issue too.

---

Well I think that this is kind of related, and somewhat the same issue, but just another manifestation of it?
I just think we may want to find a watertight solution for all these kinds of problems, as these libraries are bundled in the ndk (`libcxx` is just the most obvious one).

Out of curiosity I've tried:

```
          --prefix LD_LIBRARY_PATH : $out/libexec/android-sdk/ndk/${package.revision}/toolchains/llvm/prebuilt/linux-x86_64/lib
```

in line 67 instead. This also seems to fix the issue, and may be even more correct, as it uses the libraries in defined in the ndk (thus potentially fewer problems with version mismatch?) (apart from the hardcoded `linux-x86_64` platform).
But I'm not sure if it's the best solution...

---

Please fill a full issue to clarify what's the error message you are getting? When is it happening? What's your setup? What is your expectation?

---

The issue is the same as in #298285 just swapped `libc++.so.1` with `libxml2.so.2` (so I think one step further than without this PR).

Well my expectation is to fix all those NDK linking issues (and not just the obvious `libcxx`).
I don't think it makes sense to open another issue, as I'm pretty sure this is the same issue as in #298285.
As suggested there, it's related to the `lib64` path that doesn't exist anymore in the ndk.

I further debugged this, and found a different solution for it.
by adding:

```
    if [ -d $out/libexec/android-sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/lib ]; then
      addAutoPatchelfSearchPath $out/libexec/android-sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/lib
    fi

    find toolchains -type d -name bin -or -name lib64 -or -name lib | while read dir; do
```

in the `patchElfBnaries` hook.

---

It's confusing, because the current changes in this PR already fixed the issue while calling `clang -v`, but you are saying you had to add one more lib there to fix it! First we have to reproduce your problem, which clearly we cannot, so you need to explain it better. What's the exact error while you use the head of this branch without changes? What are you running?

---

Yes I know, it fixes that particular issue in the reproduction of that issue, but it doesn't seem to fix the broken NDK in the way that I get android apps compiled that seem to depend on the bundled `libxml2`.

I'm trying to run the `with_winit` example of [vello](https://github.com/linebender/vello) with `cargo apk run --release -p with_winit`, I'm using the `shell.nix`  in the [nixpkgs example androidenv](https://github.com/NixOS/nixpkgs/blob/26c9d4cc4893a6bdd31dcf9f10edc97c8d6a0b84/pkgs/development/mobile/androidenv/examples/shell.nix).

Anyway I've opened #313255 as alternative.

---

Thank @Janrupf for clarifying. I confirm `ld` complained about it. after this change it's like this now:

```
$ nix-shell --run '$ANDROID_SDK_ROOT/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/bin/ld'
ld: error: no input files
```